src/adaptors/mod.rs

```diff
@@ -581,6 +581,19 @@ where
     fn size_hint(&self) -> (usize, Option<usize>) {
         (0, self.iter.size_hint().1)
     }
+
+    fn fold<B, F>(mut self, acc: B, mut f: F) -> B
+    where
+        Self: Sized,
+        F: FnMut(B, Self::Item) -> B,
+    {
+        let res = self.iter.try_fold(acc, |acc, item| match item {
+            Some(item) => Ok(f(acc, item)),
+            None => Err(acc),
+        });
+        let (Err(res) | Ok(res)) = res;
```

Comment:
You can replace this `let` with a `match` to get it to pass our MSRV check.

Reply:
Thanks a lot & Thanks for your sharing about using `try_fold` for overriding `fold`. Eye-opening!